lib/LaTeXML/Post/SVG.pm

```diff
@@ -145,104 +144,95 @@ my %converters = (    # CONSTANT
   'ltx:arc'     => \&convertArc,     'ltx:dots'  => \&convertDots);
 
 sub convertNode {
-  my ($parent, $node) = @_;
+  my ($self, $parent, $node) = @_;
   my $tag = getQName($node);
   if (!$tag) {
     $parent->appendChild($node); }
   elsif (my $converter = $converters{$tag}) {
-    &$converter($parent, $node); }
+    &$converter($self, $parent, $node); }
   else {
     # Node is random LaTeXML element, and so will need an svg:foreignObject wrapper.
     # Moreover, svg will want to know the size of the foreign thing.
     # Hopefully, this will have been recorded on a containing ltx:g, using innerwidth/innerheight.
-    my $g = $parent->addNewChild($svgURI, 'g');
-    $g->setAttribute(transform => "scale(1 -1)");
-    my $new       = $g->addNewChild($svgURI, 'foreignObject');
     my $oldparent = $node->parentNode;
-    my $width     = $node->getAttribute('width')                 # Use node's own width, if any
-      || $node->getAttribute('imagewidth')                       # or if an image
-      || $oldparent->getAttribute('innerwidth')                  # else hopefully from containing ltx:g
+    my $width     = $node->getAttribute('width')    # Use node's own width, if any
+      || $node->getAttribute('imagewidth')          # or if an image
+      || $oldparent->getAttribute('innerwidth')     # else hopefully from containing ltx:g
       || $oldparent->getAttribute('width');
     my $height = $node->getAttribute('height')
       || $node->getAttribute('imageheight')
       || $oldparent->getAttribute('innerheight')
       || $oldparent->getAttribute('height');
-    $width  = "50pt" unless defined $width;
-    $height = "20pt" unless defined $height;
-    $new->setAttribute(width    => $width);
-    $new->setAttribute(height   => $height);
+    my $depth = $node->getAttribute('depth')
+      || $oldparent->getAttribute('innerdepth')
+      || $oldparent->getAttribute('depth');
+    $width  = "1pt" unless defined $width;          # Required but must be non-zero
+    $height = "1pt" unless defined $height;
+    $depth  = "0pt" unless defined $depth;
+    my $g = $parent->addNewChild($svgURI, 'g');
+    #    my $y = to_px($height);
+    my $y = to_px($height) + to_px($depth);
+    $g->setAttribute(transform => "translate(0,$y) scale(1, -1)");
+    my $new = $g->addNewChild($svgURI, 'foreignObject');
+    $new->setAttribute(width    => to_px($width))  if defined $width;
+    $new->setAttribute(height   => to_px($height)) if defined $height;
     $new->setAttribute(overflow => 'visible');
     $new->appendChild($node); }
   return; }
 
 sub convertPath {
-  my ($parent, $node) = @_;
+  my ($self, $parent, $node) = @_;
   my $newNode = $parent->addNewChild($svgURI, 'path');
-  copy_attributes($newNode, $node);
-  map { convertNode($newNode, $_) } element_nodes($node);
+  $self->copy_valid_attributes($newNode, $node);
+  map { convertNode($self, $newNode, $_) } element_nodes($node);
   return $newNode; }
 
 sub XXXconvertPicture {
```

Comment:
time to remove the `XXX` method? Seeing how it needed an update.

Reply:
indeed